gui/src/app/Project/ProjectDataModel.ts

```diff
@@ -33,13 +33,16 @@ const validateSamplingOpts = (x: SamplingOpts): boolean => {
   if (naturalFields.some((f) => !Number.isInteger(f))) return false;
   if (positiveFields.some((f) => f <= 0)) return false;
   if (nonnegativeFields.some((f) => f < 0)) return false;
+
+  if (Object.keys(x).length !== 5) return false;
```

Comment:
Not sure I love hard-coding the 5; it isn't likely to be an issue in a serious way but I can see us tripping over this every time we change the number of potential sampling opts (which admittedly isn't going to be often).